poll/translations/fr_CA/LC_MESSAGES/text.po

```diff
@@ -0,0 +1,369 @@
+msgstr ""
+
+#. Translators: {noun} is either "Answer" or "Question". {item} identifies the
+#. answer or question.
+#: poll/poll.py:313
+#, python-brace-format
+msgid "{noun} {item} contains no key."
+msgstr ""
+
+#. Translators: {noun} is either "Answer" or "Question".
+#. {noun_lower} is the lowercase version of {noun}.
+#: poll/poll.py:325
+#, python-brace-format
+msgid ""
+"{noun} has no text or img. Please make sure all {noun_lower}s have one or "
+"the other, or both."
+msgstr ""
+
+#. Translators: {noun} is either "Answer" or "Question".
+#. {noun_lower} is the lowercase version of {noun}.
+#: poll/poll.py:336
+#, python-brace-format
+msgid ""
+"{noun} was added with no label. All {noun_lower}s must have labels. Please "
+"check the form. Check the form and explicitly delete {noun_lower}s if not "
+"needed."
+msgstr ""
+
+#: poll/poll.py:344
+msgid ""
+"All images must have an alternative text describing the image in a way that "
+"would allow someone to answer the poll if the image did not load."
+msgstr ""
+
+#. Translators: "{noun_lower} is either "answer" or "question".
+#: poll/poll.py:356
+#, python-brace-format
+msgid "You must include at least one {noun_lower}."
+msgstr ""
+
+#: poll/poll.py:404
+msgid "Maximum Submissions missing or not an integer."
+msgstr ""
+
+#: poll/poll.py:409
+msgid "Private results may not be False when Maximum Submissions is not 1."
+msgstr ""
+
+#: poll/poll.py:446 poll/poll.py:838
+msgid "Poll"
+msgstr ""
+
+#: poll/poll.py:447
+msgid "What is your favorite color?"
+msgstr ""
+
+#: poll/poll.py:452
+msgid "Red"
+msgstr ""
+
+#: poll/poll.py:453
+msgid "Blue"
+msgstr ""
+
+#: poll/poll.py:454
+msgid "Green"
+msgstr ""
+
+#: poll/poll.py:455
+msgid "Other"
+msgstr "Autre"
+
+#: poll/poll.py:457
+msgid "The answer options on this poll."
+msgstr ""
+
+#: poll/poll.py:461 poll/poll.py:864
+msgid "Total tally of answers from students."
+msgstr ""
+
+#: poll/poll.py:462
+msgid "The student's answer"
+msgstr ""
+
+#: poll/poll.py:670 poll/poll.py:1143
+msgid "You have already voted in this poll."
+msgstr ""
+
+#: poll/poll.py:675
+msgid "Answer not included with request."
+msgstr ""
+
+#. Translators: {choice} uniquely identifies a specific answer belonging to a
+#. poll or survey.
+#: poll/poll.py:684
+#, python-brace-format
+msgid "No key \"{choice}\" in answers table."
+msgstr ""
+
+#: poll/poll.py:693 poll/poll.py:1151
+msgid "You have already voted as many times as you are allowed."
+msgstr ""
+
+#: poll/poll.py:723
+msgid "You must specify a question."
+msgstr ""
+
+#: poll/poll.py:726 poll/poll.py:823 poll/poll.py:1205 poll/poll.py:1317
+msgid "Answer"
+msgstr "Réponse"
+
+#: poll/poll.py:822 poll/poll.py:1206 poll/poll.py:1316
+msgid "Question"
+msgstr "Question"
+
+#: poll/poll.py:824 poll/poll.py:1318
+msgid "Submissions count"
+msgstr ""
+
+#: poll/poll.py:835
+msgid "Survey"
+msgstr "Sondage"
+
+#: poll/poll.py:841
+msgid "Yes"
+msgstr "Oui"
+
+#: poll/poll.py:842
+msgid "No"
+msgstr "Non"
```

Comment:
@OmarIthawi  I'm not seeing any of these translated strings showing up in Studio/LMS when I use this XBlock?  I have:
* installed your branch
* added `fr-ca` to the list of [LANGUAGES available on the platform](https://github.com/edx/edx-platform/blob/155a32c7972312cb6a3c309163d3e943a80b136f/lms/envs/common.py#L1234)
* visited /update_lang and entered `fr-ca`

But I'm still seeing Yes/No in the Survey and "Other" in the Poll demo content.

How did you test this?

Reply:
That's right. Unfortunately this is an expected issue with all XBlocks. XBlock fields are never translated. So for now we can only translate the following in the XBlocks:

 - Strings that are used inside a function and deferred after a `request` have happened
 - Strings inside HTML templates
 - Strings in the JavaScript files

Yes it a pretty strange issue that I get surprised repeatedly by.

So all the gettext usage for fields like `display_name = String(default=_('Survey'))` is effectively a placeholder until someone figures out a solution.

What do you think?

Reply:
@OmarIthawi

> Strings that are used inside a function and deferred after a request have happened

Hmm.. this works for https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/156  -- maybe because it uses a [separate `default_data` module](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/master/drag_and_drop_v2/default_data.py) for its data instead of using class variables?

> Strings inside HTML templates
> Strings in the JavaScript files

Ah ok, and since there aren't any `fr-CA` strings translated for these type of files, then there's not really a good way to verify this change?

Reply:
Thanks @pomegranited. I will double check the following:

 - The `default_data` module in DnDv2
 - Do more tests and screenshots for the `fr-CA`